### DISCUSSION TITLE
Error installing mpi4py due to configuration linking MPI programs

### DISCUSSION THREAD

**Discussion by git-tea** [mpi4py/mpi4py#236] [Q&A] [answered]
Hello,

I am getting the following error when building mpi4py both from source and via pip:
error: Cannot compile MPI programs. Check your configuration!!!

The pip install command I am using to produce this is: env MPICC=/opt/anaconda3/bin/mpicc pip install mpi4py

I have also run the command without the mpicc specification, as well as with other mpicc specifications, and it produces the same error.

I am running on Ubuntu 22.04, and the version of MPI being linked in the command above corresponds to Mpich 3.3.2.  I am also running this with python version 3.9.12.

The text in the following file is the full output of the command after running it:
[mpi4py_install_err.txt](https://github.com/mpi4py/mpi4py/files/9199117/mpi4py_err.txt)

Is there another configuration that I need to make to install mpi4py?

**Comment by dalcinl** [comment, marked as answer]
You are trying to build mpi4py using an MPI implementation installed in an Anaconda environment. 
It looks like you are also using spack in your system,  I see some Open MPI stuff from spack in your logs. 
What do you want do do exactly?

Do you just what mpi4py in your conda environment? Then run `conda install mpi4py`.
If you really want to install mpi4py from sources within your conda environment, then follow the following steps:

1. Make sure your conda environment is activated. Running `which mpicc` should output the mpicc command from your conda environment
2. Run `conda install mpich-mpicc`, this will install within the conda environment a compiler package. Alternatively, if you what to use your system compiler, do something like `export MPICH_CC=gcc`.
3. Finally, `pip install mpi4py`.

**Reply by git-tea** [reply]
The `conda install mpi4py` fixed it for me.  Thank you!

**Reply by dalcinl** [reply]
https://github.com/mpi4py/mpi4py/blob/master/INSTALL.rst#using-conda
